In [1]:
import pandas as pd

In [2]:
from clean_crash_data import crash_data

c:\Users\Ern\Documents\projects\cincy_traffic\clean_crash_data.py:10: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always setting a new array. To retain the old behavior, use either `df[df.columns[i]] = newvals` or, if columns are non-unique, `df.isetitem(i, newvals)`
  crash_data.loc[:, 'CRASHDATE'] = pd.to_datetime(crash_data.CRASHDATE)
c:\Users\Ern\Documents\projects\cincy_traffic\clean_crash_data.py:11: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always setting a new array. To retain the old behavior, use either `df[df.columns[i]] = newvals` or, if columns are non-unique, `df.isetitem(i, newvals)`
  crash_data.loc[:, 'DATECRASHREPORTED'] = pd.to_datetime(crash_data.DATECRASHREPORTED)
c:\Users\Ern\Documents\projects\cincy_traffic\clean_crash_data.py:18: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inpla

# Comparing Glenway to Population

## Crash Incidents

In [20]:
crash_data.CRASHDATE.min()

Timestamp('2010-02-23 17:37:00')

In [21]:
crash_data.CRASHDATE.max()

Timestamp('2023-04-05 23:30:16')

In [3]:
crash_data.describe()

LATITUDE_X   LONGITUDE_X            AGE  CRASHSEVERITY  INJURIES  \
count  3.291840e+05  3.291820e+05  288537.000000       329205.0  328908.0   
mean   2.632513e+03 -1.196852e+04      37.458738       3.484051  2.668007   
std    1.468320e+06  6.812364e+06      16.502905         1.0809  1.837925   
min   -8.449005e+01 -3.908552e+09       0.000000            1.0       1.0   
25%    3.911738e+01 -8.454953e+01      24.000000            3.0       1.0   
50%    3.913663e+01 -8.451503e+01      34.000000            3.0       1.0   
75%    3.916096e+01 -8.448475e+01      49.000000            5.0       5.0   
max    8.424329e+08  9.900000e-04     123.000000            5.0       5.0   

       MANNEROFCRASH  ROADCONTOUR  ROADSURFACE      ROADCLASS  \
count       329193.0     329193.0     329193.0  203584.000000   
mean        4.262575      1.46932     1.853129       2.825753   
std         2.441699     0.857184     0.408109       1.483882   
min              1.0          1.0          1.0       0.000000   
25%              2.0          1.0          2.0       2.000000   
50%              6.0          1.0          2.0       2.000000   
75%              6.0          2.0          2.0       4.000000   
max              9.0          9.0          9.0      10.000000   

       LIGHTCONDITIONS  ROADCONDITIONS  
count         329193.0        329193.0  
mean          1.795035        1.551853  
std           1.336298        4.993987  
min                1.0             1.0  
25%                1.0             1.0  
50%                1.0             1.0  
75%                3.0             1.0  
max                9.0            99.0

In [4]:
crash_data.columns

Index(['ADDRESS_X', 'LATITUDE_X', 'LONGITUDE_X', 'AGE',
       'COMMUNITY_COUNCIL_NEIGHBORHOOD', 'CPD_NEIGHBORHOOD',
       'SNA_NEIGHBORHOOD', 'CRASHDATE', 'CRASHLOCATION', 'CRASHSEVERITY',
       'CRASHSEVERITYID', 'DATECRASHREPORTED', 'DAYOFWEEK', 'GENDER',
       'INJURIES', 'INSTANCEID', 'LOCALREPORTNO', 'MANNEROFCRASH',
       'ROADCONTOUR', 'ROADSURFACE', 'ROADCLASS', 'ROADCLASSDESC', 'UNITTYPE',
       'TYPEOFPERSON', 'WEATHER', 'ZIP', 'CRASHSEVERITYDESCR',
       'ROADCONTOURDESCR', 'MANNEROFCRASHDESCR', 'TYPEOFPERSONDESCR',
       'LIGHTCONDITIONS', 'LIGHTCONDITIONSDESCR', 'ROADCONDITIONS',
       'ROADCONDITIONSDESCR', 'ROADSURFACEDESCR', 'INJURIESDESCR', 'STREET'],
      dtype='object')

In [7]:
crash_data[['ADDRESS_X', 'LOCALREPORTNO']].head()

ADDRESS_X LOCALREPORTNO
0  16XX HARRISON AVE     235003837
1  16XX HARRISON AVE     235003837
2  16XX HARRISON AVE     235003837
3   11XX SYCAMORE ST     235003836
4   11XX SYCAMORE ST     235003836

In [16]:
incidents_by_block = crash_data[['ADDRESS_X', 'LOCALREPORTNO']].groupby('ADDRESS_X').nunique().sort_values('LOCALREPORTNO',ascending=False)

In [18]:
incidents_by_block.describe()

LOCALREPORTNO
count   13255.000000
mean       12.825726
std        30.413601
min         1.000000
25%         1.000000
50%         3.000000
75%        10.000000
max       700.000000

In [19]:
incidents_by_block.head()

LOCALREPORTNO
ADDRESS_X                        
W MITCHELL AVE                700
2XX W MITCHELL AVE            576
1XX W MITCHELL AVE            484
61XX GLENWAY AVE              472
2XX CALHOUN ST                405

In [26]:
incidents_by_block[incidents_by_block.LOCALREPORTNO < 472].sum() / incidents_by_block.LOCALREPORTNO.sum()

LOCALREPORTNO    0.986871
dtype: float64

61XX GLENWAY AVE has 472 crashes from 2010-02-23 to 2023-04-05, whereas 75% of all street blocks have had at most 10 crashes in that same timeframe. 61XX GLENWAY AVE has had more crashes than 98.6871% of any streets.

# Glenway

In [3]:
glenway_data = crash_data[crash_data.ADDRESS_X.str.contains('GLENWAY', na=False)]